### PR TITLE
Types 2.0: Move UUID.js and add type definitions for npm package uuid 

### PR DESCRIPTION
**Pull request by iamolivinius**
Related discussion #10766 

case 1. Add a new type definition.
- [x] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [x] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [x] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url http://api.jquery.com/html .
  - it has been reviewed by a DefinitelyTyped member.

### CODE REVIEW

**Comment by andy-ms** [comment]
It's unfortunate that we've published `@types/uuid` containing definitions that don't correspond to the npm package `uuid`. I think what you've done is the best solution: move the old content to a new folder and break current users (if any) of `@types/uuid`. CC @mhegazy @RyanCavanaugh

File: uuid/index.d.ts [hunk 1]
```diff
@@ -1,82 +1,37 @@
-// Type definitions for UUID.js v3.3.0
-// Project: https://github.com/LiosK/UUID.js
-// Definitions by: Jason Jarrett <https://github.com/staxmanade/>
+// Type definitions for uuid v2.0.3
+// Project: https://github.com/defunctzombie/node-uuid
+// Definitions by: Oliver Hoffmann <https://github.com/iamolivinius/>
```
Comment (mhegazy) [inline uuid/index.d.ts]: please do not remove original authors. you can add your name to the list.

Comment (mhegazy) [inline uuid/index.d.ts]: nevermind. yes. you should remove the author for the other package.

**Comment by mhegazy** [comment]
Given that we have already published a `@types\uuid@v3.0.0` we can not just publish this one with `v2.0.3`. we need to talk to npm folks to unpublish the existing ones first. @andy-ms will try to talk to them  about this.

**Comment by andy-ms** [comment]
Moved these commits to #11785 and merged.

**Pull request closed without merging**